src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(FIND_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = (mapToBook(resultSet));
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot find book by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(FIND_ALL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> result = new ArrayList<>();
+            while (resultSet.next()) {
+                result.add(mapToBook(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Retrieving all books from DB was failed", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        String.format("Book with id=%d wasn't updated, no rows affected.",
+                                book.getId()));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new RuntimeException(
+                    String.format("Updating book with id=%d was failed", book.getId()), e
+            );
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(DELETE_QUERY)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Deletion book by id=%d was failed", id), e);
+        }
+    }
+
+    private Book mapToBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong("id");
```

Comment:
Not fixed 
<img width="988" alt="image" src="https://github.com/mate-academy/jv-jdbc-intro/assets/96740815/6d89721f-4cdd-4963-ab54-e6e6ff40ea52">